dace/transformation/dataflow/trivial_map_elimination.py

```diff
@@ -48,12 +49,16 @@ def apply(self, graph, sdfg):
 
         if len(remaining_ranges) == 0:
             # Redirect map entry's out edges
+            write_only_map = False
             for edge in graph.out_edges(map_entry):
                 path = graph.memlet_path(edge)
                 index = path.index(edge)
 
-                # Add an edge directly from the previous source connector to the destination
-                graph.add_edge(path[index - 1].src, path[index - 1].src_conn, edge.dst, edge.dst_conn, edge.data)
+                if len(path) > 1:
```

Comment:
Technically, you could also use [`Memlet.is_empty`](https://github.com/spcl/dace/blob/f99bbcabdf993d5c969094a8c803bf6037e835b5/dace/memlet.py#L218), which indicates that the edge does not carry/read data.